src/lib/remark-plugin-adjust-link-urls.ts

```diff
@@ -1,15 +1,43 @@
+import path from 'path'
 import { visit } from 'unist-util-visit'
 import { Link } from 'mdast'
 import { Plugin, Transformer } from 'unified'
 
+const getIsInternalPath = (url: string) => {
+	try {
+		new URL(url)
+		return false
+	} catch (e) {
+		return true
+	}
+}
+
 const remarkPluginAdjustLinkUrls: Plugin = ({
+	currentPath = '',
 	urlAdjustFn,
 }: {
+	currentPath: string
 	urlAdjustFn: (url: string) => string
 }): Transformer => {
 	return function transformer(tree) {
 		visit(tree, ['link', 'definition'], (node: Link) => {
-			node.url = urlAdjustFn(node.url)
+			let urlToAdjust = node.url
+
+			/**
+			 * This is to handle links like "variables/input", which are effectively
+			 * the same thing as "./input" in the context of its current page.
+			 */
+			const isInternalPath = getIsInternalPath(node.url)
+			const needsPreAdjustment = isInternalPath && /^[./?#]/.test(node.url)
```

Comment:
Maybe I'm reading this wrong but this would only match `./` links right? so the 'variables/inputs' example wouldn't hit? Also is it trying to match anchor links?

Reply:
Ahh I forgot the `!` 😂 so that it does _not_ match links that start with `.` (includes `./` and `../`), `/`, `?`, or `#`

Reply:
This is fixed now!

Reply:
oh awesome! 👍🏻 😆